openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1979,6 +2011,8 @@ TCN_IMPLEMENT_CALL(void, SSL, setVerify)(TCN_STDARGS, jlong ssl, jint level, jin
            tcn_ThrowException(e, "failed to allocate tcn_ssl_verify_config_t");
            return;
        }
+       // Copy the verify depth form the context in case depth is <0.
+       verify_config->verify_depth = c->verify_config.verify_depth;
```

Comment:
This PR creates the need to call `setVerify` and ignore the `depth`. Before this PR the depth would always be set so it didn't matter.